src/modules/languages/rust.nix

```diff
@@ -113,10 +126,21 @@ in
           let
             toolchain =
               if cfg.channel == "nightly"
-              then rustPackages.latest
-              else rustPackages.${cfg.channel};
+              then
+                rustPackages.latest
+              else
+                rustPackages.${cfg.channel}
+            ;
           in
           (builtins.mapAttrs (_: pkgs.lib.mkDefault) toolchain);
+
+        packages = [
+          (rustPackages.combine
+            (
+              (map (c: config.languages.rust.toolchain.${c}) cfg.components) ++
+              (map (t: rustPackages.targets.${t}.latest.rust-std) cfg.targets)
```

Comment:
1. Would you always want the nightly channel here? Should this use the channel specified in the config instead?
2. Is the standard library the only component you would ever want to include for a given target?

Reply:
1. You are correct, we probably want whatever is supplied by the user.
2. I'm still new to rust, but as I understand it std is what you compile against and the other things in the toolchain should run on your own computer and should therefore be your arch.

Please correct me if I'm wrong.